mantidproject/mantid
# Fix non-ortho view clim autoscaling for MDHisto workspaces in sliceviewer

**Description of work.**

Use decorator/wrapper for QuadMesh with get_array_clipped method. Autoscaling of clim requires image object (in this case QuadMesh output from pcolormesh) to have method `get_array_clipped_to_bounds` (as in `SamplingImage` and `ModestImage`).

**To test:**

(1) Make a event workspace
```
md_3D = CreateMDWorkspace(Dimensions=3, Extents=[-1,1,-2,2,-0.1,0.1], 
    Names="H,K,L", Frames='HKL,HKL,HKL',Units='r.l.u.,r.l.u.,r.l.u.')
FakeMDEventData(InputWorkspace=md_3D, UniformParams='1e4') # 4D data
expt_info = CreateSampleWorkspace()
md_3D.addExperimentInfo(expt_info)
SetUB(Workspace=md_3D, c=2, gamma=120)
```
(2) Open `md_3D` in sliceviewer, zoom/pan and check color limits update if autoscale checked
(3) Switch to view orthogonal axes (e.g. H and L) - check autoscale still works
(3) Create a MDHisto workspace
```
mdHisto_2D = BinMD(InputWorkspace='md_3D', AxisAligned=False, 
    BasisVector0='H,r.l.u.,1.0,0.0,0.0', 
    BasisVector1='K,r.l.u.,0.0,1.0,0.0', 
    BasisVector2='L,r.l.u.,0.0,0.0,1.0', 
    OutputExtents='-1,1,-2,2,-0.1,0.1',
    OutputBins='20,10,1', NormalizeBasisVectors=False)
DeleteWorkspace(md_3D)
```
(5) Repeat step (2) with the `mdHisto_2D ` workspace 

Fixes #33409





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
